bss/batch_submitter.go

```diff
@@ -0,0 +1,178 @@
+package bss
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ethereum-optimism/optimistic-specs/bss/drivers/l2output"
+	"github.com/ethereum-optimism/optimistic-specs/bss/txmgr"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	hdwallet "github.com/miguelmota/go-ethereum-hdwallet"
+	"github.com/urfave/cli"
+)
+
+const (
+	// defaultDialTimeout is default duration the service will wait on
+	// startup to make a connection to either the L1 or L2 backends.
+	defaultDialTimeout = 5 * time.Second
+)
+
+// Main is the entrypoint into the batch submitter service. This method returns
+// a closure that executes the service and blocks until the service exits. The
+// use of a closure allows the parameters bound to the top-level main package,
+// e.g. GitVersion, to be captured and used once the function is executed.
+func Main(gitVersion string) func(ctx *cli.Context) error {
+	return func(ctx *cli.Context) error {
+		cfg, err := NewConfig(ctx)
+		if err != nil {
+			return err
+		}
+
+		log.Info("Initializing batch submitter")
+
+		batchSubmitter, err := NewBatchSubmitter(cfg, gitVersion)
+		if err != nil {
+			log.Error("Unable to create batch submitter", "error", err)
+			return err
+		}
+
+		log.Info("Starting batch submitter")
+
+		if err := batchSubmitter.Start(); err != nil {
+			return err
+		}
+		defer batchSubmitter.Stop()
+
+		log.Info("Batch submitter started")
+
+		<-(chan struct{})(nil)
```

Comment:
yes this will block forever. it's not super critical to add signal handling since we don't have really have anything that needs proper closing, i.e. a database connection, but added it anyways just in case 👍